foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/dao/Dao.java

```diff
@@ -0,0 +1,43 @@
+package com.boozallen.aissemble.configuration.dao;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Configuration::Store
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.configuration.exception.PropertyDaoException;
+
+import java.util.Set;
+
+public interface Dao<T> {
+
+    boolean checkEmpty();
+
+    /**
+     * Read T from Vault with given group name and property name
```

Comment:
A: Not Vault-specific

Reply:
yeah.. this has been corrected locally. Let me push it.